src/model/tls/default.rs

```diff
@@ -0,0 +1,164 @@
+extern crate rustls;
+
+use self::rustls::{
+    client::{NoClientSessionStorage, ServerCertVerified, ServerCertVerifier},
+    Certificate as RustlsCertificate, ClientConfig, ClientConnection, Error as RustlsError,
```

Comment:
Also, no need to refer to `rustls` as `self::rustls` there, I don't think.

Reply:
> Why not just use `rustls::Certificate` and the like? Doing `use foo::Bar as FooBar;` seems counterproductive.

I defined a struct `Certificate` in this file, so I had two options to prevent name collision:
- rename it using `as` keyword (current version)
- use absolute path, like rustls::Certificate (your proposition)

And in the docs: https://doc.rust-lang.org/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html#creating-idiomatic-use-paths

they say that both ways are idiomatic, but if it is confusing then I can change it to the later 😄 


> Also, no need to refer to `rustls` as `self::rustls` there, I don't think.

In this case I had to do it because this was the only place that I've introduced the external crate using `extern crate rustls;` and also this is old crate that is using `rust2015` so the use keyword worked in a different way that is is working now. 
But I can move that extern call to the root of the module, to the `lib.rs` file, and then I will be able to drop that `self::` prefix.

Reply:
Moving `extern crate rustls;` to `lib.rs` (gated by `#[cfg(not(feature = "fips"))]`?) seems to be the better option; alternatively, maybe move `extern crate boring;` to `src/model/tls/fips.rs` for consistency.

Reply:
Also, renaming `Certificate as RCertificate` or something like that could make sense, but if the disambiguous name is going to be just as long as the full path anyway, just use the full path.